### DISCUSSION TITLE
Ask for input before execution if there are placeholders

### DISCUSSION THREAD

**Discussion by SaneBow** [TheR1D/shell_gpt#204] [Show and tell]
I find that many times when I asked for a shell command, I forgot to specify every necessary concrete values, so the output command from GPT contains placeholders.
In those cases, I can't directly execute them, but need to copy the command, replace those placeholders, and paste to execute.

Why not check for placeholders and ask for concrete inputs before execution?
I've implemented it [here](https://github.com/SaneBow/shell_gpt/tree/ask_arguments).
If this is also in other people's favor, I will create a PR.

Here's how it looks like:

<img width="920" alt="sgpt_ask" src="https://user-images.githubusercontent.com/5907249/233603012-1837ed27-c6b0-445b-bea7-5d8ce59eaf74.png">

**Comment by SaneBow** [comment]
An alternative UX that support editing can be as the following:

```
Execute shell command? [y/i/e/N]
```

If user input `e`, we open the default shell editor (`vi`) to edit the command.
If user input `i`, we prompt for values for placeholders as proposed above.

**Comment by TheR1D** [comment]
Since your prompt does not specify a word or file name, GPT generates responses with placeholders. However, you can include all the necessary details in your prompt.
```shell
sgpt -s "count the occurance of 'python' in 'readme.md'"
grep -o 'python' readme.md | wc -l
```
